frontend/src/components/elements/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,176 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ComponentType, ReactElement, useRef, useState } from "react"
+import { useTheme } from "@emotion/react"
+import { Tabs as UITabs, Tab as UITab } from "baseui/tabs-motion"
+
+import { BlockNode, AppNode } from "src/lib/AppNode"
+import { useIsOverflowing } from "src/lib/Hooks"
+
+import { StyledTabContainer } from "./styled-components"
+
+const TAB_HEIGHT = "2.5rem"
+
+export interface Props {
+  widgetsDisabled: boolean
+  node: BlockNode
+  isStale: boolean
+}
+
+function Tabs(TabLayoutComponent: ComponentType<any>): ComponentType<any> {
+  const TabContainer = (props: Props): ReactElement => {
+    const { widgetsDisabled, node, isStale } = props
+
+    const [activeKey, setActiveKey] = useState("0")
+    const tabListRef = useRef<HTMLUListElement>(null)
+
+    const isOverflowing = useIsOverflowing(tabListRef, false)
+
+    const theme = useTheme()
+
+    return (
+      <StyledTabContainer
+        isOverflowing={isOverflowing}
+        tabHeight={TAB_HEIGHT}
+        className="stTabs"
+      >
+        <UITabs
+          activeKey={activeKey}
+          onChange={({ activeKey }) => {
+            setActiveKey(activeKey.toString())
+          }}
+          disabled={widgetsDisabled}
+          overrides={{
+            TabHighlight: {
+              style: () => ({
+                backgroundColor: widgetsDisabled
+                  ? theme.colors.fadedText40
+                  : theme.colors.primary,
+                height: `2px`,
+                // Requires bottom offset to align with the TabBorder
+                bottom: `3px`,
+              }),
+            },
+            TabBorder: {
+              style: () => ({
+                backgroundColor: theme.colors.fadedText05,
+                height: `2px`,
+              }),
+            },
+            TabList: {
+              props: { ref: tabListRef },
+              style: () => ({
+                gap: theme.spacing.lg,
+                ...(isStale
+                  ? {
+                      opacity: 0.33,
+                      transition: "opacity 1s ease-in 0.5s",
+                    }
+                  : {}),
+              }),
+            },
+            Root: {
+              style: () => ({
+                // resetting transform to fix full screen wrapper
+                transform: `none`,
+              }),
+            },
+          }}
+          activateOnFocus
+        >
+          {node.children.map(
+            (appNode: AppNode, index: number): ReactElement => {
+              const childProps = {
+                ...props,
+                ...{ node: appNode as BlockNode },
```

Comment:
nit: could this be written as simply
```typescript
{
  ...props,
  node: appNode as BlockNode
}
```
without the second spread operator?

Reply:
That's a copy-paste from Block.tsx. I simplified it there as well 👍